core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1576,11 +1576,11 @@ private Map<String, Argument> analyzeArguments(Node node, List<ArgumentSpecifica
                 throw semanticException(INVALID_ARGUMENTS, errorLocation, "Too many arguments. Expected at most %s arguments, got %s arguments", argumentSpecifications.size(), arguments.size());
             }
 
-            if (arguments.isEmpty()) {
+            if (argumentSpecifications.isEmpty()) {
                 return ImmutableMap.of();
             }
 
-            boolean argumentsPassedByName = arguments.stream().allMatch(argument -> argument.getName().isPresent());
+            boolean argumentsPassedByName = !arguments.isEmpty() && arguments.stream().allMatch(argument -> argument.getName().isPresent());
```

Comment:
Why do we need `!arguments.isEmpty() &&`?

Reply:
For the purpose of analysis, we need to choose if the case of no passed arguments falls under "arguments passed by name" or "arguments passed positionally". I chose the latter.

Reply:
it would maybe be more symmetric if you had variables
```
anyArgumentPassedByName
anyArgumentPassedByPosition
```